### PR TITLE
Moved camtest.js logic to camresolutiontest and improved stats reporting

### LINKED ISSUE #428: Combine Test video feed and Camera resolution tests
Currently the two tests do mostly the same thing but report different metrics, makes sense to combine the two in order to reduce test run time and boilerplate code.

### PR DESCRIPTION
**Pull request by KaptenJansson**
* Integrated camtest.js into camresolutiontest.js in order to reduce run time.
* Removed camtest.js
* Mandatory resolutions are now tested as separate test cases.

Fixes #428.

### CODE REVIEW

File: samples/web/content/testrtc/js/camresolutionstest.js [hunk 1]
```diff
@@ -7,44 +7,96 @@
  */
 'use strict';
 
-/* This test tries calling getUserMedia() with each resolution from the list
- * below. Each gUM() call triggers a success or a fail callback; we report
- * ok/nok and schedule another gUM() with the next resolution until the list
- * is exhausted. Some resolutions are mandatory and make the test fail if not
- * supported.
+ /*
+ * Analyze performance for X resolution uses getStats, canvas and the video
```
Comment (andresusanopinto) [inline samples/web/content/testrtc/js/camresolutionstest.js]: Move this paragraph next to the code that add the "Analyze performance for X" tests. Also quote the test name so it clear it is a name and not part of the sentence.

File: samples/web/content/testrtc/js/camresolutionstest.js [hunk 2]
```diff
@@ -7,44 +7,96 @@
...
+ * element to analyze the video frames from a capture device. It will report
+ * number of black frames, frozen frames, tested frames and various stats like
+ * average encode time and FPS.
+ *
+ * Supported resolution test tries calling getUserMedia() with each resolution
```
Comment (andresusanopinto) [inline samples/web/content/testrtc/js/camresolutionstest.js]: Same here... move the comment paragraph next to the line that adds the test.

File: samples/web/content/testrtc/js/camresolutionstest.js [hunk 3]
```diff
@@ -7,44 +7,96 @@
...
+                   [1920, 1200, false],
+                   [3840, 2160, false],  // 4K
+                   [4096, 2160, false]];
+
+// Loop through resolutions array and create a test case where video performance
+// is analyzed for for each mandatory resolution.
+for (var index = 0; index < resolutions.length; index++) {
+  if (resolutions[index][2]) {
+    addTest('Camera', 'Analyze performance for ' +
+            JSON.stringify(resolutions[index][0]) + 'x' +
```
Comment (andresusanopinto) [inline samples/web/content/testrtc/js/camresolutionstest.js]: No need to use JSON.stringify? int should convert to string without it....
Or is this a lint warning?

File: samples/web/content/testrtc/js/camresolutionstest.js [hunk 4]
```diff
@@ -73,38 +125,72 @@ CamResolutionsTest.prototype = {
   },
 
   successFunc_: function(stream) {
+    this.stream = stream;
     this.supportedResolutions++;
     var theResolution = this.resolutions[this.counter++];
```
Comment (andresusanopinto) [inline samples/web/content/testrtc/js/camresolutionstest.js]: Can you fix this variable name? theResolution? wtf is that name?

File: samples/web/content/testrtc/js/camresolutionstest.js [hunk 5]
```diff
@@ -120,47 +206,119 @@ CamResolutionsTest.prototype = {
...
+  },
+
+  testExpectations_: function(info) {
+    for (var key in info) {
+      if (info.hasOwnProperty(key)) {
+        reportInfo(key + ': ' + info[key]);
+      }
+    }
+
+    if (info.avgFPSSent < 5) {
```
Comment (andresusanopinto) [inline samples/web/content/testrtc/js/camresolutionstest.js]: info.agvFPSSent probably does not exists here.

File: samples/web/content/testrtc/js/camresolutionstest.js [hunk 6]
```diff
@@ -120,47 +206,119 @@ CamResolutionsTest.prototype = {
...
-      reportInfo('Sent FPS: ' + minFPSSent + ' min / ' + avgFPSSent +
-                 ' avg / ' + maxFPSSent + ' max');
-      if (avgFPSSent < 5) {
-        reportError('Low average sent FPS: ' + avgFPSSent);
-      }
+      // TODO: Add a reportInfo() function with a table format to display
+      // values clearer.
+      statsReport.actualVideoWidth = this.video.videoWidth;
+      statsReport.actualVideoHeight = this.video.videoHeight;
+      statsReport.encodeSetupTime = this.timeForFirstFrameMs + ' (ms)';
```
Comment (andresusanopinto) [inline samples/web/content/testrtc/js/camresolutionstest.js]: I don't like adding the (ms) and so on here... e.g. code further one can expect this to be an int and not a string.

Comment (andresusanopinto) [inline samples/web/content/testrtc/js/camresolutionstest.js]: Maybe suffix the variables with the units.

File: samples/web/content/testrtc/js/camresolutionstest.js (hunk 1 above)
Comment (KaptenJansson) [inline samples/web/content/testrtc/js/camresolutionstest.js]: Done!

File: samples/web/content/testrtc/js/camresolutionstest.js (hunk 2 above)
Comment (KaptenJansson) [inline samples/web/content/testrtc/js/camresolutionstest.js]: Done!

File: samples/web/content/testrtc/js/camresolutionstest.js (hunk 3 above)
Comment (KaptenJansson) [inline samples/web/content/testrtc/js/camresolutionstest.js]: Not sure why I added it, removed it and it works fine.

File: samples/web/content/testrtc/js/camresolutionstest.js (hunk 4 above)
Comment (KaptenJansson) [inline samples/web/content/testrtc/js/camresolutionstest.js]: renamed it to selectedResolution

File: samples/web/content/testrtc/js/camresolutionstest.js (hunk 6 above)
Comment (KaptenJansson) [inline samples/web/content/testrtc/js/camresolutionstest.js]: I did suffix it originally but changed my mind. Have now suffixed them again.

File: samples/web/content/testrtc/js/camresolutionstest.js (hunk 5 above)
Comment (KaptenJansson) [inline samples/web/content/testrtc/js/camresolutionstest.js]: Indeed it doesn't, good catch. Fixed.

**Pull request merged**